app/Controller/EventsController.php

```diff
@@ -88,6 +88,12 @@ function edit($id = null) {
                 unset($this->request->data['Event']['finish_time']);
             }
 
+            if(!empty($this->request->data['Event']['deadline_date'])) {
+                $this->request->data['Event']['registration_deadline'] = TimeLib::mysqlDateTimeFormat($this->request->data['Event']['deadline_date'], $this->request->data['Event']['deadline_time']);
+                unset($this->request->data['Event']['deadline_date']);
+                unset($this->request->data['Event']['deadline_time']);
+            }
```

Comment:
```
else {
   $this->request->data['Event']['registration_deadline'] = null;
}
```

to remove a deadline previously added?

Reply:
Good catch, hadn't noticed this.  PR incoming.